### PR TITLE
Generate Mermaid from a State Machine

### PR DESCRIPTION
**Pull request by Synesso**
Generate mermaid diagrams given a starting state

### CODE REVIEW

**Review by hugomd — APPROVED** [review]
Nice!

File: README.md [hunk 1]
```diff
@@ -65,6 +65,13 @@ from a given starting state.
 StateMachine.verify(Awake).isRight() shouldBe true
 ```
 
+#### Document your state machine
+
+The utility `StateMachine.mermaid` will generate a mermaid diagram of your state machine. This can be rendered in markdown.
```
Comment (rgrnwd) [inline README.md:70]: That is useful.

File: README.md [hunk 2]
```diff
@@ -65,6 +65,13 @@ from a given starting state.
...
+The diagram of HamsterStates above was created using this utility.
+
+```kotlin
```
Comment (elliot-smith) [inline README.md:73]: Is this missing some example code?

Comment (rgrnwd) [inline README.md:73]: Do you need closing tags here?

Comment (rgrnwd) [reply README.md:73]: It looks a little bit broken. 
Screenshot 2024-01-25 at 3 34 45 pm

Comment (rgrnwd) [reply README.md:73]: +1

Comment (Synesso) [reply README.md:73]: TY

**Pull request merged**